compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -9683,8 +9696,6 @@ OMR::Z::TreeEvaluator::loadaddrEvaluator(TR::Node * node, TR::CodeGenerator * cg
                   reloType = TR_ClassAddress;
                else if (node->getSymbol()->isMethod())
```

Comment:
In `OMR::Z::TreeEvaluator::checkAndSetMemRefDataSnippetRelocationType` you changed 
```
if (cg->comp()->compileRelocatableCode())
 ...
else if (node->getSymbol()->isClassObject())
```
into
```
else if (cg->needClassAndMethodPointerRelocations() && node->getSymbol()->isClassObject())
```
I see a similar pattern here. Should we change this too?

Reply:
Functionally the `compileRelocatableCode()` check is correct. For consistency (in z as well as other codgens) I think we can change it.

Reply:
I've improved the checks here: d3a28b6f017b9a6147392196f3ecd4ca7b15faf2.

I'm going to mark this PR as a WIP while I wait for some tests to pass. Will update once ready. Otherwise this is still good for review.